docs/hardening_guide.md

```diff
@@ -0,0 +1,403 @@
+# Hardening Guide
+
+This document provides prescriptive guidance for hardening a production installation of RKE2. It outlines the configurations and controls required to address Kubernetes benchmark controls from the Center for Information Security (CIS).
+
+If RKE2 is run in CIS mode, RKE2 applies a restrictive NetworkPolicy and PodSecurityPolicy. The restrictive NetworkPolicy allows for only namespace traffic with the exception of DNS and applies to `kube-system`, `kube-public`, and `default` namespaces. The restrictive PodSecurityPolicy addresses CIS controls defined in section 5.2. More details can be found below.
```

Comment:
I'd like "in CIS mode" replaced with "with the `profile` flag set to `cis-1.5`.
I'd also like us to enumerate everything that happens when you turn on this flag.

Reply:
Remediated.